bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -406,6 +413,14 @@ synchronized BookieId sendNextRead() {
 
             try {
                 BookieId to = ensemble.get(bookieIndex);
+                if (clientCtx.getBookieWatcher().isBookieUnavailable(to)) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Skipping send read request to unavailable bookie: {}", to.getId());
+                    }
+                    erroredReplicas.set(replica);
+                    sentReplicas.set(replica);
+                    return null;
```

Comment:
We need to trigger the next read. Otherwise, if this is the first read request, no more reads will be sent.

Reply:
We could add a configuration, whether allowing to skip read from unavailable bookie